### PR TITLE
Respect port from topic URI

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Addresses issue mentioned in https://github.com/pmcilreavy/AzureEventGridSimulator/issues/22#issuecomment-788010913

### CODE REVIEW

**Comment by pakrym** [comment]
I don't see any tests verifying the port is included in the request. What am I missing?

**Comment by pakrym** [comment]
>I don't see any tests verifying the port is included in the request. What am I missing?

I'm missing all those Asserts in tests :)

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs [hunk 1]
```diff
@@ -299,5 +299,44 @@ public async Task SerializesExpectedProperties_DerivedType()
...
+            Assert.AreEqual(5, cloudEvent.Data.ToObjectFromJson<DerivedTestPayload>().DerivedProperty);
+
+            List<CloudEvent> eventsList = new List<CloudEvent>()
+            {
+                cloudEvent
+            };
+
+            await client.SendEventsAsync(eventsList);
+
+            Assert.AreEqual(5, cloudEvent.Data.ToObjectFromJson<DerivedTestPayload>().DerivedProperty);
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:339]: Why are we asserting something that was asserted 2 lines before?

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs [hunk 2]
```diff
@@ -299,5 +299,44 @@ public async Task SerializesExpectedProperties_DerivedType()
...
+            };
+            EventGridPublisherClient client =
+               new EventGridPublisherClient(
+                   new Uri("https://contoso.com:100/api/events"),
+                   new AzureKeyCredential("fakeKey"),
+                   options);
+            var cloudEvent = new CloudEvent(
+                    "record",
+                    "Microsoft.MockPublisher.TestEvent",
+                    new DerivedTestPayload
```
Comment (pakrym) [inline sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:323]: Do we need the extra-fancy payload for this test?

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:339]: Eh, we are running the cloudevents through the publisher which could mutate them with extensions.

**Review by pakrym — APPROVED** [review]

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:323]: probably not, but I didn't see the harm in it.

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:339]: Seems like that should have it's own test.

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:339]: We do have tests for that.

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs (hunk 2 above)
Comment (pakrym) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:323]: With time it becomes harder to understand what's important for the scenario being tested and what isn't.

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:323]: Well, the test name should help with that, but I see your point.

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs:339]: Then I would trim unrelated asserts from these ones.

**Comment by pmcilreavy** [comment]
I tested with 4.0.0-beta.4 and it did not work (as expected) and then I tested with 4.0.0-beta.5 and was successfully able to send an event to my Azure Event Grid Simulator. So yay 🎉

**Comment by JoshLove-msft** [comment]
Hmm, you tested with the published beta.5 from nuget? The fix hasn't been released yet and would appear in the dev feeds as an alpha I believe.

**Comment by pmcilreavy** [comment]
Sorry my bad. I initially tried with 4.0.0-alpha.20210304.1 which worked (event received by the simulator below). I then changed to 4.0.0-beta.5 and it seemed like that also worked. I just re-tested and yes,
beta.5 does not work. I guess I didn't clean or restore nuget properly perhaps. Sorry not sure what happened - user error. 
Anyway, apologies for the confusion.

[
  {
    "id": "eb32aeec-32ed-42ee-8956-d2505e6af2a2",
    "subject": "/the/subject/4.0.0-alpha.20210304.1+e55b6a1fadeff56090f0eaaeeaae2b495eee0b57",
    "data": {
      "Id": 1
    },
    "eventType": "The.Event.Type",
    "eventTime": "2021-03-05T22:28:49.4033463Z",
    "dataVersion": "v1",
    "metadataVersion": "1",
    "topic": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/eventGridSimulator/providers/Microsoft.EventGrid/topics/AzureMessagingEventGridClientTest"
  }
]
Test harness attached
[GridTest.zip](https://github.com/Azure/azure-sdk-for-net/files/6093873/GridTest.zip)

**Comment by JoshLove-msft** [comment]
No worries at all! Thanks for verifying the fix @pmcilreavy!

**Pull request merged**